src/OrchardCore.Modules/OrchardCore.Users/Views/UserFields.Edit.cshtml

```diff
@@ -47,13 +50,74 @@
 
     <script at="Foot" asp-name="userFieldsPassword">
         document.addEventListener('DOMContentLoaded', function() {
-            const strongPasswordLength = 12;
             let password = document.getElementById('@Html.IdFor(p => p.Password)');
             let passwordConfirmation = document.getElementById('@Html.IdFor(p => p.PasswordConfirmation)');
             let passwordToggle = document.getElementById('passwordToggle');
             let passwordConfirmationToggle = document.getElementById('passwordConfirmationToggle');
             let generate = document.getElementById('generatePassword');
             let copy = document.getElementById('copyPassword');
+            const requiredPasswordLength = @(Math.Max(IdentityOptions.Value.Password.RequiredLength, IdentityOptions.Value.Password.RequiredUniqueChars));
+            const allowed = {
+                uppers: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+                lowers: 'abcdefghijklmnopqrstuvwxyz',
+                digits: '0123456789',
+                symbols: '!@@#$%^&*'
```

Comment:
Is it taken from ASP.NET identity code?

Reply:
No. These are just the symbols available of the number keys on the keyboard. Microsoft asp.net core allow any special characters without the need to list all available characters. If it is not digit or a-saw-Z character, it’s considered a special characters. Here we just need a list of special characters to randomly pick from. We could add more special characters but I feel what we have there is enough to generate the password securely assuming that the identity password option is set to long enough value.

Reply:
No. These are just the symbols available of the number keys on the keyboard. Microsoft asp.net core allow any special characters without the need to list all available characters. If it is not digit or a-saw-Z character, it’s considered a special characters. Here we just need a list of special characters to randomly pick from. We could add more special characters but I feel what we have there is enough to generate the password securely assuming that the identity password option is set to long enough value.

Reply:
Perhaps Seb needs to ensure that's align with identity system

Reply:
If `RequireNonAlphanumeric` is true and there is a character that is not a letter or a digit it’ll pass The framework’s validation 
https://github.com/dotnet/aspnetcore/blob/36df8e5dc487e27cb85674ef3f144dca07f05c20/src/Identity/Extensions.Core/src/PasswordValidator.cs#L55